src/Logging/src/DynamicSerilog/SerilogWebHostBuilderExtensions.cs

```diff
@@ -4,44 +4,91 @@
+    /// Enables to configure Serilog from code instead of configuration.
+    /// </param>
+    /// <returns>
+    /// The incoming <see cref="IWebHostBuilder" />, so that additional calls can be chained.
+    /// </returns>
+    public static IWebHostBuilder AddDynamicSerilog(this IWebHostBuilder hostBuilder, Action<WebHostBuilderContext, LoggerConfiguration>? configureLogger)
+    {
+        return AddDynamicSerilog(hostBuilder, configureLogger, false);
+    }
+
+    /// <summary>
+    /// Adds Serilog with Console sink, wrapped in a <see cref="DynamicSerilogLoggerProvider" />.
+    /// </summary>
+    /// <param name="hostBuilder">
+    /// The <see cref="IWebHostBuilder" /> to configure.
     /// </param>
     /// <param name="preserveDefaultConsole">
-    /// When true, do not remove Microsoft's ConsoleLoggerProvider.
+    /// When set to <c>true</c>, does not remove existing logger providers.
     /// </param>
     /// <returns>
-    /// The <see cref="IWebHostBuilder" />.
+    /// The incoming <see cref="IWebHostBuilder" />, so that additional calls can be chained.
     /// </returns>
-    public static IWebHostBuilder AddDynamicSerilog(this IWebHostBuilder hostBuilder, Action<WebHostBuilderContext, LoggerConfiguration> configureLogger = null,
-        bool preserveDefaultConsole = false)
+    public static IWebHostBuilder AddDynamicSerilog(this IWebHostBuilder hostBuilder, bool preserveDefaultConsole)
     {
-        return hostBuilder.ConfigureLogging((hostContext, logBuilder) =>
+        return AddDynamicSerilog(hostBuilder, null, preserveDefaultConsole);
+    }
+
+    /// <summary>
+    /// Adds Serilog with Console sink, wrapped in a <see cref="DynamicSerilogLoggerProvider" />.
+    /// </summary>
+    /// <param name="hostBuilder">
+    /// The <see cref="IWebHostBuilder" /> to configure.
+    /// </param>
+    /// <param name="configureLogger">
+    /// Enables to configure Serilog from code instead of configuration.
+    /// </param>
+    /// <param name="preserveDefaultConsole">
+    /// When set to <c>true</c>, does not remove existing logger providers.
+    /// </param>
+    /// <returns>
+    /// The incoming <see cref="IWebHostBuilder" />, so that additional calls can be chained.
+    /// </returns>
+    public static IWebHostBuilder AddDynamicSerilog(this IWebHostBuilder hostBuilder, Action<WebHostBuilderContext, LoggerConfiguration>? configureLogger,
+        bool preserveDefaultConsole)
+    {
+        ArgumentGuard.NotNull(hostBuilder);
+
+        return hostBuilder.ConfigureLogging((hostContext, loggingBuilder) =>
         {
-            LoggerConfiguration loggerConfiguration = null;
+            LoggerConfiguration? loggerConfiguration = null;
 
             if (configureLogger != null)
             {
-                loggerConfiguration = new LoggerConfiguration().ReadFrom.Configuration(hostContext.Configuration);
```

Comment:
Thanks for asking, great question. I'm not entirely sure about this. In fact, I struggled to understand how this is supposed to work.

At first, I assumed that passing `configureLogger` would _supplement_ what was loaded from appSettings. But [here](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Logging/src/DynamicSerilog/SerilogLoggingBuilderExtensions.cs#L36-L46) (which calls either [this](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Logging/src/DynamicSerilog/SerilogOptions.cs#L54) or [this](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Logging/src/DynamicSerilog/SerilogOptions.cs#L27)), the settings from `IConfiguration` are explicitly discarded when `configureLogger` is used. If calling `configureLogger` would be additive, I'd expect the code [here](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Logging/src/DynamicSerilog/SerilogHostBuilderExtensions.cs#L37-L41) to also load the configuration in the `else` clause, but it doesn't.

Also, the tests at https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Logging/test/DynamicSerilog.Test/SerilogDynamicWebhostBuilderTest.cs seem to suggest it's either loading from configuration _or_ setting up through code.

Because of the above, I ultimately inferred the intended purpose was to setup either through code or through configuration.

Reply:
Bart you are correct. It's not both, it's either or.